lib/peep.rb

```diff
@@ -1,13 +1,39 @@
 require_relative 'database_connection'
 
 class Peep
+  
+  attr_reader :id, :content, :created_at
+
+  def initialize(id:, content:, created_at:)
+    @id = id
+    @content = content
+    @created_at = created_at # automatically get CURRENT TIMESTAMP
+  end
+
   def self.all
     rs = DatabaseConnection.query("SELECT * FROM peeps ORDER BY id DESC;")
 
-    rs.map { |peep| peep['content'] }
+    array_of_peeps = []
+
+    rs.each do |peep_data|  # rs (result object) is NOT A REAL PEEP instance just data entries. we grab these from the database with each
+      peep = Peep.new(                                    # use the data entries (rs) to create a Peep
+        id: peep_data['id'],
+        content: peep_data['content'],
+        created_at: Time.parse(peep_data['created_at'])  # Takes a string representation of a Time and attempts to parse it
```

Comment:
Very good use of Time module. I had no idea .parse automatically gets the current timestamp. well done!

Reply:
Thank's! It works well. but didn't manage to feature test this properly so far